### ISSUE TITLE
No lowercase on key name in bulkloader

### ISSUE THREAD

**Issue by tucotuco** [VertNet/Darwin-Core-Engine#48]
Right now the source_id is lowercased before using it to create the key name. Since this is the only record we have of the source_id, it should not be altered.

**Comment by eightysteele** [comment]
Yeah, so in thinking about this some more, `Record` should probably have a couple of new properties like `source_id_column`, `source_id_value`, and maybe even `source_filename`.

**Comment by tucotuco** [comment]
Don't think any of that is necessary. Need a compelling use case. But, definitely need an unaltered source_id somewhere.

**Comment by eightysteele** [comment]
> Don't think any of that is necessary. Need a compelling use case. But, definitely need an unaltered source_id somewhere.

Right, so a publisher downloads record annotations and now needs to link them back to their source data. If they've published records with different `source_id` columns, then they'll need to know which `source_id_column` belongs to which record so that they can make the join on that column with the `source_id_value`.

**Comment by tucotuco** [comment]
Not a realistic scenario. source_id is the collection-level unique identifier, by definition - the closest thing to stable information in the source data. If they are changing source_ids - all bets are off, and knowing what file and field it came from originally doesn't help to re-integrate annotations with the original data.

**Comment by eightysteele** [comment]
So @tucotuco and I face-to-faced this issue a bit. The basic use case to support here is publishing CSV files that are generated from different data sources having different `source_id` columns. This comes into play when publishers want to link these records in VertNet back to their data sources. I think we agree that it's unclear how useful this would be. So, we're tabling the issue for a bit while we think on it and get some feedback from others on the team.

**Comment by tucotuco** [comment]
+1
